File: day7/1.rb
#Read file
data = File.readlines('data.txt').sample.strip.split(',').map(&:to_i)

#Get the median
data = data.sort
mid = data[data.length/2]

avg = (data.reduce(:+).to_f / data.length) #Get avg for task 2


sum1 = 0
sum2 = 0
sum3 = 0
#Loop through array to get the fuel counts
for spot in data
    sum1 += (spot - mid).abs
    sum2 += (1..(spot - avg.floor).abs).sum
    sum3 += (1..(spot - avg.ceil).abs).sum
end

puts "Task 1:"
p sum1
puts "Task 2:"
p [sum2, sum3].min

File: 2015/day01/1.py
import os

floor = 0
char_pos = 0

below_zero = 0

with open('data.txt') as f:
    while True:
        char = f.read(1)
        if not char:
            break
        char_pos += 1
        if char == ')':
            floor -= 1
        if char == '(':
            floor += 1

        if floor < 0 and below_zero == 0:
            below_zero = char_pos

print('Floor ended on:')
print(floor)

print('Basement entered on:')
print(below_zero)